src/pages/docs/platform/architecture/infrastructure-operations.mdx

```diff
@@ -0,0 +1,208 @@
+
+When an instance starts, it's configured with specific roles to run and associated container image versions. This configuration determines which services the instance will provide and how they will interact with the rest of the cluster.
+
+### Service discovery
+
+The services running on each instance advertise themselves to the common gossip-based discovery layer. When services start, they register their capabilities with the discovery system, making themselves available to other components that need to use them.
+
+The gossip protocol ensures all nodes share an eventually-consistent view of available services across the cluster. This distributed approach eliminates single points of failure in the service discovery process and provides resilience against network partitions and node failures.
+
+Other services can discover and communicate with appropriate instances based on their advertised capabilities and health status. If a service becomes unhealthy, it's automatically removed from the registry, ensuring clients are only directed to functioning services.
+
+This service discovery mechanism enables the platform to adapt quickly to infrastructure changes, including scaling events and instance failures, without manual intervention and is a key component of Ably's self-healing capabilities.
+
+### Inbound connectivity
+
+Inbound connectivity from internet-connected client devices to each site is handled through a multi-layered approach designed for reliability and performance.
+
+AWS CloudFront serves as the outermost layer, providing DDoS protection and global distribution through Amazon's extensive edge network. This ensures clients can connect to a nearby edge location, reducing latency.
+
+Behind CloudFront, a collection of Network Load Balancers (NLBs) distribute connections among frontend instances. NLBs operate at Layer 4 (transport layer) and can handle millions of connections with very low latency, making them ideal for WebSocket traffic.
+
+DNS-based routing with latency-based resolution directs clients to the nearest available region. This ensures optimal performance under normal conditions while maintaining the ability to route traffic away from problematic regions.
+
+If a region becomes unavailable, clients can connect to alternative regions through fallback mechanisms built into Ably's client libraries. This ensures service continuity even during regional outages.
+
+## Configuration management
+
+Ably employs a two-pronged approach to managing its infrastructure configuration, combining infrastructure as code for static resources with a dynamic service configuration system for runtime parameters.
+
+### Infrastructure as code
+
+Terraform is used to manage the resources provisioned for AWS and all other cloud services providers. This means that the entire layout of the infrastructure and much of its configuration is managed as code, providing numerous benefits for operational reliability.
```

Comment:
I still think we should use generic IAC terminology here as opposed to specifically referencing Terraform

Reply:
Removed all mention of Terraform ce9aa7cf7d9565166113b4b6560a6f8bc27bc794